src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilsTest.java

```diff
@@ -143,4 +152,159 @@ public void testGetAccessModifierFromModifiersTokenWithNullParameter() {
+        final DetailAST notSetterMethod = setterMethod.getNextSibling();
+
+        assertTrue(CheckUtils.isSetterMethod(setterMethod));
+        assertFalse(CheckUtils.isSetterMethod(notSetterMethod));
+    }
+
+    @Test
+    public void testGetAccessModifierFromModifiersToken() throws Exception {
+        final DetailAST privateVariable = getNodeFromFile(TokenTypes.VARIABLE_DEF);
+        final DetailAST protectedVariable = privateVariable.getNextSibling();
+        final DetailAST publicVariable = protectedVariable.getNextSibling();
+        final DetailAST packageVariable = publicVariable.getNextSibling();
+
+        assertEquals(AccessModifier.PRIVATE,
+                CheckUtils.getAccessModifierFromModifiersToken(privateVariable.getFirstChild()));
+        assertEquals(AccessModifier.PROTECTED,
+                CheckUtils.getAccessModifierFromModifiersToken(protectedVariable.getFirstChild()));
+        assertEquals(AccessModifier.PUBLIC,
+                CheckUtils.getAccessModifierFromModifiersToken(publicVariable.getFirstChild()));
+        assertEquals(AccessModifier.PACKAGE,
+                CheckUtils.getAccessModifierFromModifiersToken(packageVariable.getFirstChild()));
+    }
+
+    @Test
+    public void testGetFirstNode() throws Exception {
+        final DetailAST classDef = getNodeFromFile(TokenTypes.CLASS_DEF);
+
+        assertEquals(classDef.getFirstChild().getFirstChild(), CheckUtils.getFirstNode(classDef));
+    }
+
+    @Test
+    public void testIsReceiverParameter() throws Exception {
+        final DetailAST objBlock = getNodeFromFile(TokenTypes.OBJBLOCK);
+        final DetailAST methodWithReceiverParameter = objBlock.getLastChild().getPreviousSibling();
+        final DetailAST receiverParameter =
+                getNode(methodWithReceiverParameter, TokenTypes.PARAMETER_DEF);
+        final DetailAST simpleParameter =
+                receiverParameter.getNextSibling().getNextSibling();
+
+        assertTrue(CheckUtils.isReceiverParameter(receiverParameter));
+        assertFalse(CheckUtils.isReceiverParameter(simpleParameter));
+    }
+
+    @Test
+    public void testParseDouble() throws Exception {
+        assertEquals(1.0, CheckUtils.parseDouble("1", TokenTypes.NUM_INT), 0);
+        assertEquals(-0.05, CheckUtils.parseDouble("-0.05f", TokenTypes.NUM_FLOAT), 0);
+        assertEquals(8.0, CheckUtils.parseDouble("8L", TokenTypes.NUM_LONG), 0);
+        assertEquals(0.0, CheckUtils.parseDouble("0.0", TokenTypes.NUM_DOUBLE), 0);
+        assertEquals(2915.0, CheckUtils.parseDouble("0x0B63", TokenTypes.NUM_INT), 0);
+        assertEquals(2.147_483_647E10,
+                CheckUtils.parseDouble("21474836470", TokenTypes.NUM_LONG), 0);
+        assertEquals(59.0, CheckUtils.parseDouble("073L", TokenTypes.NUM_LONG), 0);
+    }
+
+    private static File getFile(String filename) {
+        return new File("src/test/resources/com/puppycrawl/tools/checkstyle/utils/" + filename);
```

Comment:
See https://github.com/checkstyle/checkstyle/issues/3891 for the naming convention of directories for inputs.